heat/core/logical.py

```diff
@@ -307,3 +314,55 @@ def logical_xor(t1, t2):
     tensor([ False, False,  True])
     """
     return operations.__binary_op(torch.logical_xor, t1, t2)
+
+
+def _sanitize_close_input(x, y):
+    if np.isscalar(x):
+        try:
+            if isinstance(y, dndarray.DNDarray):
+                x = factories.array(float(x), device=y.device)
+            else:
+                x = factories.array(float(x))
+        except (ValueError, TypeError):
+            raise TypeError("Data type not supported, input was {}".format(type(x)))
+
+    elif not isinstance(x, dndarray.DNDarray):
+        raise TypeError(
+            "Only tensors and numeric scalars are supported, but input was {}".format(type(x))
+        )
+
+    if np.isscalar(y):
```

Comment:
Code duplication, factor out in a separate function

Reply:
Right, thanks, I've rewritten this part